pkg/api/controller.go

```diff
@@ -4016,7 +4016,6 @@ func (c *Controller) OtfDiff(w http.ResponseWriter, r *http.Request, repository,
 
 	entries, err := c.otfDiffService.RunDiff(ctx, params.Type, tdp)
 	if err != nil {
-		c.Logger.Error(err)
```

Comment:
I think we _should_ log this an error here!  `err` is often much more detailed than what we will hand the client.

Consider reporting with something like
```go
c.Logger.WithError(err).WithField("tablediff_params", fmt.Sprintf("%+v", tdp.TablePaths)).WithField("repo", repo).Error("Diff service failed")

Reply:
done

Reply:
Actually we should also `.WithContext(ctx)`, which links in the request and other future goodies.